mlir/include/mlir/Dialect/Rock/IR/RockAcceptingViewOpInterface.td

```diff
@@ -0,0 +1,61 @@
+//===- RockAcceptingViewOpInterface.td - ops that accept rock views---------===//
+//
+// Part of the rocMLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2022 Advanced Micro Devices INc.
+//===----------------------------------------------------------------------===//
+//
+// This file defines RockAcceptingViewOpInterface, which abstracts rock operations
+// that accept a view as operands.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef ROCK_ACCEPTINGVIEWOP_INTERFACE
+#define ROCK_ACCEPTINGVIEWOP_INTERFACE
+
+include "mlir/IR/OpBase.td"
+
+def RockAcceptingViewOpInterface : OpInterface<"RockAcceptingViewOpInterface"> {
+  let description = [{
+    Any rock operation that accepts a view as input should implement this interface.
+  }];
+  let cppNamespace = "::mlir::rock";
+
+  let methods = [
+    InterfaceMethod<
+        /*desc=*/[{
+          "Clone the operation with a new set of extra indices"
+        }],
+        /*retType=*/"Operation *",
+        /*methodName=*/"cloneWithExtraIndices",
+        /*args=*/(ins "OpBuilder& ":$builder, "Value":$view, "ArrayRef<Value>":$newExtraIndices),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/""
+      >,
+    InterfaceMethod<
+        /*desc=*/[{
+          "Return the extra indices used to index the operand accepting the view"
+        }],
+        /*retType=*/"::mlir::Operation::operand_range",
+        /*methodName=*/"getExtraIndices",
+        /*args=*/(ins),
+        /*methodBody=*/"",
+        /*defaultImplementation=*/""
+      >,
+    InterfaceMethod<
```

Comment:
I am not sure I follow. Operations like threadwise_writeall have an interface like (raw_buffer, view, extraIndicesForTheView, additionalView,...) . Are you suggesting to return an OpOperand *operand to view and get the extraIndices as operand++?

Reply:
(writing this interface was a bit painful anyway, so I am happy to better nail it down)

Reply:
Do you need to do a `++` to get to the extraIndices at all?

If not, I'd just go for `SmallVectorImpl<OpOperand*>&` (or `OpOperand`) that's a list of the operands that've been viewed

Reply:
Yes, but that would get me the operands that accept views. I need the indices passed to the view. Basically the interface will be used like:
```
originalIndices = interface.getExtraIndices();
newIndices = originaIndices + additionalIndices;
newOp = interface.cloneWithNewIndices(newIndices, newView);
```

Reply:
Or are you saying that I can get the extra indices from the view?
```
SmallVectorImpl<OpOperand*>& operands = interface.getOperandsAcceptingView();
originaIndices = /*get indices from the operands*/;
newIndices = originaIndices + additionalIndices;
newOp = interface.cloneWithNewIndices(newIndices, newView);
```

Reply:
Hold on, backing up, is it true that there's only one viewed argument that has extra indices?

Reply:
Yes, for now this is the case. 
- `threadwise_readInto/threadwise_writeAll` accept a single set of extra indices
- `threadwise_copy` does not accept any extra index
But I am happy to generalize this for a generic operation with a signature like: `(operand0, extraIndices0, ...., operandN-1, extraIndicesN-1)` : an idea could be:
```
originalIndices = interface.getExtraIndices(OperandIndex);
newIndices = originaIndices + additionalIndices;
newOp = interface.cloneWithNewIndices(OperandIndex, newIndices, newView);
```
But I am happy to hear more thoughts

Reply:
Ok, so if there's only one operation, then it's as `OperandRange` for the extra indices an an `OpOperand` for the viewed value

Reply:
Yes, and this is why the `getExtraindices` returns an `operand_range`. I don't understand why I need the `OpOperand`

Reply:
The point is that, instead of referring to operands by index, you should return the relevant `OpOperand`. Among other things, this'll save you an `O(n)` scan for which operand is the one that has the views applied to it

Reply:
Alright, done(I think)!